backend/de.metas.business/src/main/java/de/metas/product/impexp/ProductImportProcess.java

```diff
@@ -283,6 +273,11 @@ private void createUpdateProductPrice(final I_I_Product imp)
 		pp.setM_PriceList_Version_ID(priceListVersionId);    // FK
 		pp.setM_Product_ID(productId.getRepoId()); // FK
 		pp.setC_TaxCategory_ID(taxCategoryId);
+		if (X_I_Product.INVOICABLEQTYBASEDON_CatchWeight.equals(imp.getInvoicableQtyBasedOn()))
+		{
+			pp.setInvoicableQtyBasedOn(imp.getInvoicableQtyBasedOn());
+			productImporter.handleCatchWeight(imp);
```

Comment:
why we can't handle catch weight in this class?

Reply:
I cand, but in importer helper, did , in general any other complex logic